src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static removeFromAlive(animal) {
+    Animal.alive = Animal.alive.filter((a) => a !== animal);
+  }
+
+  static getAlive() {
+    return Animal.alive.map(({ name, health, hidden }) => ({
+      name,
+      health,
+      hidden,
+    }));
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        Animal.removeFromAlive(animal);
+      }
+    }
```

Comment:
The bite method in the Carnivore class should include a check to ensure that the animal being bitten is not another Carnivore. Currently, the method only checks if the animal is an instance of Herbivore and not hidden. It should also include an else condition to handle the case where a Carnivore attempts to bite another Carnivore, which the method should not allow.